runners/java-fn-execution/src/test/java/org/apache/beam/runners/fnexecution/EmbeddedSdkHarness.java

```diff
@@ -37,15 +37,18 @@
 import org.apache.beam.sdk.fn.stream.OutboundObserverFactory;
 import org.apache.beam.sdk.options.PipelineOptionsFactory;
 import org.apache.beam.vendor.guava.v32_1_2_jre.com.google.common.util.concurrent.ThreadFactoryBuilder;
+import org.junit.Rule;
 import org.junit.rules.ExternalResource;
 import org.junit.rules.TestRule;
+import org.junit.rules.Timeout;
 
 /**
  * A {@link TestRule} which creates a {@link FnHarness} in a thread, services required for that
  * {@link FnHarness} to properly execute, and provides access to the associated client and harness
  * during test execution.
  */
 public class EmbeddedSdkHarness extends ExternalResource implements TestRule {
+  @Rule public transient Timeout globalTimeout = Timeout.seconds(600);
```

Comment:
Rather than adding this everywhere, can we do this in a more targeted way?

Reply:
e.g. do we know which test should be timing out?

Reply:
Yes this is inconvenience. junit by defult does not have a timeout, and while gradle can setup per task timeout, I do not find a way to setup a per-test timeout with build.gradle or gradle parameters.

Reply:
> e.g. do we know which test should be timing out?

We do not know and that is the obstacle for investigation. When the workflow timeout it stops half way and all-but-the-timed-out task reports the test result. That's why I was able to find it is runners:java-fn-execution:test timing out but not able to find which test is timing out

Reply:
Got it, thanks